widgets/text/text.go

```diff
@@ -108,6 +109,25 @@ func (t *Text) Write(text string, wOpts ...WriteOption) error {
 	if opts.replace {
 		t.reset()
 	}
+
+	// If set, limit the buffer (if maxContent has been set)
+	if len(t.content) >= t.opts.maxContent && t.opts.maxContent > 0 {
+		fmt.Printf("len: %d strlen: %d", len(t.content), len(t.content))
+		// If the new content is longer than all existing, simply reset
+		if len(text) >= len(t.content) {
+			t.reset()
+		} else {
+			// Truncate by the space required for the new entry
+			t.content = t.content[runewidth.StringWidth(text):]
```

Comment:
The calls to `runewidth.StringWidth` do take some work (at the very least O(n) scan over the text. Any way we could refactor this so that the code calls `runewidth.StringWidth(text)` only once?

Reply:
Unless my logic is bad, this should only be called twice during the whole Write procedure. I haven't profiled how intensive it is so if it's worth calling once then happy for this to be refactored.